kotlinx-coroutines-core/native/src/Dispatchers.kt

```diff
@@ -4,18 +4,21 @@
 
 package kotlinx.coroutines
 
+import kotlinx.coroutines.internal.multithreadingSupported
 import kotlin.coroutines.*
+import kotlin.native.concurrent.*
 
 /** Not inside [Dispatchers], as otherwise mutating this throws an `InvalidMutabilityException`. */
+@ThreadLocal
```

Comment:
Do I understand correctly that, if `Dispatchers.setMain` was used, then `withContext (Dispatchers.Main)` will behave differently depending on from which thread it's called? I don't think such behavior is desirable.

Reply:
Right, this is a good catch. I've moved it back to the object